src/dxvk/dxvk_instance.cpp

```diff
@@ -105,6 +105,26 @@ namespace dxvk {
           extensionsEnabled))
       throw DxvkError("DxvkInstance: Failed to create instance");
 
+    // Hide VK_EXT_debug_utils behind an environment variable. This extension
+    // adds additional overhead to winevulkan
+    if (env::getEnvVar("DXVK_PERF_EVENTS") == "1") {
+      std::array<DxvkExt*, 1> utilsExtList = {{
+        &insExtensions.extDebugUtils,
+      }};
+
+      DxvkNameSet utilsExtensionsEnabled;
+
+      if (!extensionsAvailable.enableExtensions(
+            utilsExtList.size(),
+            utilsExtList.data(),
+            utilsExtensionsEnabled))
+        throw DxvkError("DxvkInstance: Failed to create instance");
+
+      extensionsEnabled.merge(utilsExtensionsEnabled);
+    }
+
+    m_extensions = insExtensions;
```

Comment:
I think a much easier way to do this would be to turn `insExtensionList` into an `std::vector` and conditionally push the pointer to it. The whole extension merging thing is meant for extensions that we don't necessarily know about.

Reply:
Okay that makes sense. So the merging is for the "plugins" like for OpenVR/OpenXR or DXVK Extensions like dxvk-ags or dxvk-nvapi ? Either way, fixed.